CMakeLists.txt

```diff
@@ -8,8 +8,10 @@ if(POLICY CMP0074)
 endif()
 
 # Set target C++ standard and required compiler features
-set(CMAKE_CXX_STANDARD 14 CACHE STRING "The target C++ standard. PCL requires C++14 or higher.")
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "The target C++ standard. PCL requires C++17 or higher.")
```

Comment:
Why did you change the C++ standard?

Reply:
Curiosity. 

You seemed to indicate in [this issue](https://github.com/PointCloudLibrary/pcl/issues/4083) that you were OK with moving to C++17, so I thought I'd experiment and see how CI would handle it.